framework/common/WindowContext.kt

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (C) 2022 Vaticle
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+package com.vaticle.typedb.studio.framework.common
+
+import androidx.compose.ui.awt.ComposeWindow
+
+class WindowContext(val height: Int, val width: Int, val x: Int, val y: Int) {
+    constructor(window: ComposeWindow): this(window.height, window.width, window.x, window.y)
+}
```

Comment:
As discussed, instead of mocking we refactor Studio itself to include the use of a WindowContext class that makes itself available to aspects of Studio trying to make decisions based on the dimensions and position of the window on the screen.